### PR TITLE
page/your_account: send asset: show balance 

### PR DESCRIPTION
**Pull request by tharvik**
* show balance in "send asset" dialog
* fix dialog not showing the right user
* cargo deny ignore `chrono`'s RUSTSEC
* split dialog in dedicated file

### CODE REVIEW

File: web/src/agents/node/get_balance.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+use at2_node::client::{self, Client};
+use at2_ns::ThinUser;
+use wasm_bindgen_futures::spawn_local;
+use yew::worker::*;
+
+use crate::config::Config;
+
+/// Get the balance of an user
```
Comment (ineiti) [inline web/src/agents/node/get_balance.rs]:
```suggestion
/// Get the balance of a user
```

the binding `n` is not used here as it's pronounced `yuser`, and the `y` doesn't ask for a binding `n`...

https://english.stackexchange.com/questions/105116/is-it-a-user-or-an-user

File: web/src/agents/node/get_balance.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+impl Agent for GetBalance {
+    type Reach = Context<Self>;
+    type Message = (HandlerId, Result<u64, client::Error>);
+    type Input = ThinUser;
+    type Output = Result<u64, client::Error>;
+
+    fn create(link: AgentLink<Self>) -> Self {
+        let conf = Config::parse().unwrap(); // TODO unwrap
```
Comment (ineiti) [inline web/src/agents/node/get_balance.rs]: Resolve TODOs, also on line 25.

File: web/src/pages/your_account/mod.rs [hunk 3]
```diff
@@ -70,29 +66,23 @@ impl Component for YourAccount {
...
+            Self::Message::SendTransaction((recipient, amount)) => {
                 let sequence = 1; // TODO retrieve latest sequence
 
-                self.send_asset_agent
-                    .send((user, sequence, recipient, amount as u64));
+                self.send_asset_agent.send((
+                    self.props.user.clone(),
+                    sequence,
+                    recipient,
+                    amount as u64,
```
Comment (ineiti) [inline web/src/pages/your_account/mod.rs]: Is that a safe assumption here? Or is `amount` user-input and might be tainted?

File: web/src/pages/your_account/send_transaction_dialog.rs [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    dialog::{ActionType, MatDialogAction},
+    MatButton, MatDialog, MatFormfield, MatList, MatListItem, WeakComponentLink,
+};
+use yew::{prelude::*, worker::Agent};
+
+use super::transaction_builder::TransactionBuilder;
+use crate::agents;
+
+#[derive(Properties, Clone, PartialEq)]
+pub struct Properties {
```
Comment (ineiti) [inline web/src/pages/your_account/send_transaction_dialog.rs:15]: Please add some short comments for each of the public structures. Like `golint` which is expecting to have a comment for all exported elements.

**Review by ineiti — CHANGES_REQUESTED** [review]
From a syntactic point of view it looks good, except the couple of comments.

File: web/src/agents/node/get_balance.rs (hunk 1 above)
Comment (tharvik) [reply web/src/agents/node/get_balance.rs]: English is weird some times, thanks for the link :)

File: web/src/pages/your_account/mod.rs (hunk 3 above)
Comment (tharvik) [reply web/src/pages/your_account/mod.rs]: It is for the target we are using: `amount` is a `usize`, which is "on a 32 bit target, this is 4 bytes and on a 64 bit target, this is 8 bytes.".

https://doc.rust-lang.org/std/primitive.usize.html

Adding a TODO for now to handle it when fixing all the TODOs.

File: web/src/agents/node/get_balance.rs (hunk 2 above)
Comment (tharvik) [reply web/src/agents/node/get_balance.rs]: Most TODO resolved and `unwrap` handled.

File: web/src/agents/node/get_balance.rs (hunk 1 above)
Comment (ineiti) [reply web/src/agents/node/get_balance.rs]: s/some times/sometimes/ ;)

**Review by ineiti — APPROVED** [review]

**Pull request merged**